src/SignalR/clients/cpp/src/signalrclient/default_websocket_client.cpp

```diff
@@ -21,30 +21,73 @@ namespace signalr
         : m_underlying_client(create_client_config(signalr_client_config))
     { }
 
-    pplx::task<void> default_websocket_client::connect(const std::string& url)
+    void default_websocket_client::start(std::string url, transfer_format, std::function<void(std::exception_ptr)> callback)
     {
-        return m_underlying_client.connect(utility::conversions::to_string_t(url));
+        m_underlying_client.connect(utility::conversions::to_string_t(url))
+            .then([callback](pplx::task<void> task)
+            {
+                try
+                {
+                    task.get();
+                    callback(nullptr);
+                }
+                catch (...)
+                {
+                    callback(std::current_exception());
```

Comment:
I'm not convinced this pattern for Try/Catch is good. Why not catch a std::exception& and pass the ref into the callback? That way you don't need to rethrow to get the exception.